xen-bugtool

```diff
@@ -1592,7 +1596,7 @@ def fd_usage(cap):
                         fd_dict[num_fds] = []
                     fd_dict[num_fds].append(name.replace('\0', ' ').strip())
         except:
-            output += 'Error: Pid %d disappeared\n' % (d,)
+            output += "Error: Pid %s disappeared\n" % d
```

Comment:
`% (d, )` is an intentional pattern to cause the % operator to not explode when `d` happens to be a tuple.

Reply:
As a reference, I just triggered this bug while writing unit-tests as regular user (due to not having permission as user to access the fds of pid 1):
```py
IOError reading /proc/pagetypeinfo: [Errno 13] Permission denied: '/proc/pagetypeinfo'
Traceback (most recent call last):
  File "/home/bkaindl/git/github/status-report/tests/unit/../../xen-bugtool", line 1619, in fd_usage
    num_fds = len(os.listdir(os.path.join('/proc/'+d+'/fd')))
PermissionError: [Errno 13] Permission denied: '/proc/1/fd'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/bkaindl/git/github/status-report/tests/unit/../../xen-bugtool", line 632, in collect_data
    s = no_unicode(v["func"](cap))
  File "/home/bkaindl/git/github/status-report/tests/unit/../../xen-bugtool", line 1625, in fd_usage
    output += 'Error: Pid %d disappeared\n' % (d,)
TypeError: %d format: a real number is required, not str
```